test/services/supported_ssg_updater_test.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+# A class to test the SupportedSsgUpdater service
+class SupportedSsgUpdaterTest < ActiveSupport::TestCase
+  context 'fallback file' do
+    setup do
+      @supported_file = ::Tempfile.new('suported_ssg_file')
+      @supported_file.write(File.read('test/fixtures/files/supported_ssg_test.yaml'))
+      @supported_file.rewind
+
+      @fallback_file = ::Tempfile.new('fallback_ssg_file')
+    end
+
+    should 'contain correct yaml representation' do
+      SsgConfigDownloader.stub_const(:DS_FILE_PATH, @supported_file.path) do
+        SsgConfigDownloader.stub_const(:DS_FALLBACK_PATH, @fallback_file.path) do
+          updater = SupportedSsgUpdater.new
+          updater.update_ssg_fallback_yaml
+          fallback_ssg = YAML.safe_load(@fallback_file.read)
+          assert_not_empty fallback_ssg
+        end
+      end
+    end
+
+    should 'not contain unwanted keys' do
+      SsgConfigDownloader.stub_const(:DS_FILE_PATH, @supported_file.path) do
+        SsgConfigDownloader.stub_const(:DS_FALLBACK_PATH, @fallback_file.path) do
+          updater = SupportedSsgUpdater.new
+          updater.update_ssg_fallback_yaml
+
+          fallback = YAML.safe_load(@fallback_file.read)
+
+          assert_nothing_raised do
+            updater.search_through(fallback) do |key, _value, _current_hash|
+              raise 'Unwanted key' if SupportedSsgUpdater::KEYS_TO_REMOVE.include?(key)
+            end
+          end
+        end
+      end
+    end
+
+    should 'not contain any keys in profiles (except old_names)' do
+      SsgConfigDownloader.stub_const(:DS_FILE_PATH, @supported_file.path) do
+        SsgConfigDownloader.stub_const(:DS_FALLBACK_PATH, @fallback_file.path) do
+          updater = SupportedSsgUpdater.new
+          updater.update_ssg_fallback_yaml
+
+          fallback = YAML.safe_load(@fallback_file.read)
+
+          def sub_keys?(profile)
+            profile&.instance_of(Hash) && profile.keys.any? && profile.keys.exclude?('old_names')
+          end
+
+          assert_nothing_raised do
+            updater.search_through(fallback) do |key, value, _current_hash|
```

Comment:
```suggestion
            SupportedSsgUpdater.search_through(fallback) do |key, value, _current_hash|
```

but I think this should be covered by the tests above somehow.